### DISCUSSION TITLE
Volume integration over "half" of my mesh

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#27823] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello,

I want to only volume integrate an AuxVariable half of my mesh. I found InternalVolume which seems to do what I would like... but all of the examples define a region with a boundary/sideset. I would like:

Integrate this quantity from -xmin to xmax, -ymin to ymax and 0 to zmax (or zmin to 0)

ParaView can do this quite fast by using the Clip filter and the selecting the IntegrateVariables filter but I want to track this quantity over 1000s of time steps. Making a subdomain is not possible because I am using tets and it will be rather tricky to pick out the nodes that sit around the z = 0.0 point.

Any ideas? Thanks!

**Comment by GiudGiud** [comment]
Hello

if you do NOT want to add a new block to the simulation, then this is what the MeshDivision system is for.

Create a mesh division that cuts the mesh in two region
https://mooseframework.inl.gov/source/meshdivisions/CartesianGridDivision.html
Use this VPP to integrate separately over the two regions
https://mooseframework.inl.gov/source/vectorpostprocessors/MeshDivisionFunctorReductionVectorPostprocessor.html

Set the variable as the functor. 

Guillaume

**Reply by mangerij** [reply]
Guillaume, this works with tets?

**Reply by GiudGiud** [reply]
It works if the tets are aligned with the half boundary

**Reply by GiudGiud** [reply]
If they are not you can use any other type of mesh division. Something like Function bins with a ParsedFunction is probably able to catch other simple subdivisions
